app/ui/resources.tsx

```diff
@@ -106,30 +117,104 @@ export function InitCodeblock({
+      onToggle={() => setIsMenuOpen((oldValue) => !oldValue)}
+    >
+      <summary
+        className="_no-triangle absolute top-0 grid"
         data-copied={copied}
-        className="outline-none"
       >
-        {/* had to put these here instead of as a mask so we could add an opacity */}
-        <svg
-          aria-hidden
-          className="h-5 w-5 text-gray-500 hover:text-black dark:text-gray-400 dark:hover:text-gray-100"
-          viewBox="0 0 24 24"
+        <span
+          data-code-block-copy
+          data-copied={copied}
+          className={`absolute right-0 top-0 opacity-0 hover:opacity-100 ${copied || isMenuOpen ? "!opacity-100" : ""}`}
         >
-          {copied ? (
-            <use href={`${iconsHref}#check-mark`} />
-          ) : (
-            <use href={`${iconsHref}#copy`} />
-          )}
-        </svg>
-        <span className="sr-only">Copy code to clipboard</span>
-      </button>
-    </div>
+          <svg
+            aria-hidden
+            className="h-5 w-5 text-gray-500 hover:text-black dark:text-gray-400 dark:hover:text-gray-100"
+            viewBox="0 0 24 24"
+          >
+            {copied ? (
+              <use href={`${iconsHref}#check-mark`} />
+            ) : (
+              <use href={`${iconsHref}#copy`} />
+            )}
+          </svg>
+          <span className="sr-only">Copy code to clipboard</span>
+        </span>
+      </summary>
+      <DetailsPopup
+        className="!left-auto !right-0 top-10"
+        childrenClassName="!w-[110px]"
+      >
+        <Form preventScrollReset replace className="flex flex-col">
+          <input
+            type="hidden"
+            name="category"
+            value={searchParams.get("category") || ""}
+          />
+
+          <PackageManagerButton
+            packageManager="npm"
+            setCopied={(copied) => setCopied(copied, "npm")}
+          />
+          <PackageManagerButton
+            packageManager="yarn"
+            setCopied={(copied) => setCopied(copied, "yarn")}
+          />
+          <PackageManagerButton
+            packageManager="pnpm"
+            setCopied={(copied) => setCopied(copied, "pnpm")}
+          />
+          <PackageManagerButton
+            packageManager="bun"
+            setCopied={(copied) => setCopied(copied, "bun")}
+          />
+        </Form>
+      </DetailsPopup>
+    </DetailsMenu>
+  );
+}
+
+type PackageManagerButtonProps = {
+  packageManager: PackageManager;
+  setCopied: (copied: boolean) => void;
+};
+
+function PackageManagerButton({
+  packageManager,
+  setCopied,
+}: PackageManagerButtonProps) {
+  const submit = useSubmit();
+  return (
+    <button
+      className="rounded-md p-1.5 text-left text-sm hover:cursor-pointer hover:bg-gray-50"
+      type="submit"
+      onClick={() => {
+        submit({
+          preventScrollReset: false,
+        });
+        setCopied(true);
+      }}
+    >
+      {packageManager}
+    </button>
   );
 }
```

Comment:
I'm not sure I understand why you're using a `Form` or `useSubmit`. This should just be be a simple `onClick` call on a button, there's no reason for a network request since nothing happens on the client.

Also, I think you can move a lot of the logic into `PackageManagerButton` and just render something like

```tsx
["npm", "yarn", "pnpm", "bun"].map(packageManager => (
  <PackageManagerButton packageManager={packageManager} initCommand={initCommand} />
))
```